src/hotspot/share/opto/loopopts.cpp

```diff
@@ -789,7 +789,6 @@ Node *PhaseIdealLoop::conditional_move( Node *region ) {
   assert(!bol->is_OpaqueInitializedAssertionPredicate(), "Initialized Assertion Predicates cannot form a diamond with Halt");
   if (bol->is_OpaqueTemplateAssertionPredicate()) {
     // Ignore Template Assertion Predicates with OpaqueTemplateAssertionPredicate nodes.
-    assert(assertion_predicate_has_loop_opaque_node(iff), "must find OpaqueLoop* nodes");
```

Comment:
I don't think we are required to do this verification here since we are now always verifying these nodes when creating and cloning them. Here we just want to bail out if we find a Template Assertion Predicate.